### PR TITLE
Add output in robot frame option

### PR DESCRIPTION
**Pull request by bhomberg**
We found that to use Cartographer for mapping/localization, we needed to have the IMU data output not in its default NED frame or in ENU frame, but that we needed it in robot frame (x forward, y left, z up).  This PR adds an option for robot frame output and does not change the NED or ENU output.

### CODE REVIEW

**Comment by tonybaltovski** [comment]
The idea of a separate robot frame is kind of odd considering [REP-0103](https://www.ros.org/reps/rep-0103.html#coordinate-frame-conventions).  However, is it that you use the orientation without the magnetometer?

**Comment by bhomberg** [comment]
The robot frame we're using is the one specified in the link you sent, with x forward, y left, z up.  We need the output in that frame, not relative to magnetic north (if that's what you mean by using orientation without the magnetometer).  We're using [Cartographer](https://github.com/googlecartographer/cartographer_ros) for slam, which expects IMU output in robot frame; ie, for a 2D mobile robot, only minor changes in pitch/roll and yaw is equivalent to robot orientation.

**Comment by tonybaltovski** [comment]
I was trying to determine in the conditions when you would want to use the robot only frame.  Not using the magnetometer would have been one case where all data would be from the `robot_frame`. The orientation estimate from the IMU uses an EKF to fuse the magnetometer with the accelerometer and gyroscope. Thus disabling it would the orientation would be purely in the robot's frame based on its starting location.  However, this might be considered in a different PR where disabling the mag automatically places the orientation into the robot's frame.  How about changing the flag from `tf_ned_to_robot_frame` to `orientation_in_robot_frame` or something along those line?

File: src/main.cpp [hunk 1]
```diff
@@ -51,6 +51,11 @@ const char VERSION[10] = "0.0.2";   // um7_driver version
 // us to publish everything we have.
 const uint8_t TRIGGER_PACKET = DREG_EULER_PHI_THETA;
 
+enum OutputAxes
```
Comment (tonybaltovski) [inline src/main.cpp]: Let's follow the [roscpp style guide](http://wiki.ros.org/CppStyleGuide#Enumerations)

File: src/main.cpp [hunk 2]
```diff
@@ -51,6 +51,11 @@ const char VERSION[10] = "0.0.2";   // um7_driver version
...
+{
+    DEFAULT, ENU, ROBOT_FRAME
```
Comment (tonybaltovski) [inline src/main.cpp:58]: linting, two spaces please.

File: src/main.cpp [hunk 3]
```diff
@@ -382,7 +457,22 @@ int main(int argc, char **argv)
 
   // Enable converting from NED to ENU by default
   bool tf_ned_to_enu;
+  bool tf_ned_to_robot_frame;
   private_nh.param<bool>("tf_ned_to_enu", tf_ned_to_enu, true);
+  private_nh.param<bool>("tf_ned_to_robot_frame", tf_ned_to_robot_frame, false);
+  OutputAxes axes = DEFAULT;
+  if (tf_ned_to_enu && tf_ned_to_robot_frame)
+  {
+      ROS_ERROR("Requested IMU data in two separate frames.");
```
Comment (tonybaltovski) [inline src/main.cpp]: linting, two spaces please.

File: src/main.cpp [hunk 4]
```diff
@@ -382,7 +457,22 @@ int main(int argc, char **argv)
...
+  }
+  else if (tf_ned_to_enu)
+  {
+      axes = ENU;
```
Comment (tonybaltovski) [inline src/main.cpp]: linting, two spaces please.

File: src/main.cpp [hunk 5]
```diff
@@ -382,7 +457,22 @@ int main(int argc, char **argv)
...
+  }
+  else if (tf_ned_to_robot_frame)
+  {
+      axes = ROBOT_FRAME;
```
Comment (tonybaltovski) [inline src/main.cpp]: linting, two spaces please.

**Review by tonybaltovski — CHANGES_REQUESTED** [review]

**Comment by bhomberg** [comment]
Comments addressed -- ready for another look!

**Review by tonybaltovski — APPROVED** [review]

**Comment by tonybaltovski** [comment]
Awesome! Thank you @bhomberg!! Let me know if you need a release.

**Pull request merged**